[2023-10-23T17:47:06.672+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: update_dwh.sat_employee_job_details_update manual__2023-10-23T17:44:02.944833+00:00 [queued]>
[2023-10-23T17:47:06.707+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: update_dwh.sat_employee_job_details_update manual__2023-10-23T17:44:02.944833+00:00 [queued]>
[2023-10-23T17:47:06.709+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 1
[2023-10-23T17:47:06.727+0000] {taskinstance.py:1382} INFO - Executing <Task(PostgresOperator): sat_employee_job_details_update> on 2023-10-23 17:44:02.944833+00:00
[2023-10-23T17:47:06.735+0000] {standard_task_runner.py:57} INFO - Started process 4428 to run task
[2023-10-23T17:47:06.740+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'update_dwh', 'sat_employee_job_details_update', 'manual__2023-10-23T17:44:02.944833+00:00', '--job-id', '232', '--raw', '--subdir', 'DAGS_FOLDER/update_dwh_dag.py', '--cfg-path', '/tmp/tmph3i0f9ou']
[2023-10-23T17:47:06.742+0000] {standard_task_runner.py:85} INFO - Job 232: Subtask sat_employee_job_details_update
[2023-10-23T17:47:06.798+0000] {task_command.py:416} INFO - Running <TaskInstance: update_dwh.sat_employee_job_details_update manual__2023-10-23T17:44:02.944833+00:00 [running]> on host 1f8bb8a692af
[2023-10-23T17:47:06.909+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='update_dwh' AIRFLOW_CTX_TASK_ID='sat_employee_job_details_update' AIRFLOW_CTX_EXECUTION_DATE='2023-10-23T17:44:02.944833+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2023-10-23T17:44:02.944833+00:00'
[2023-10-23T17:47:06.911+0000] {sql.py:274} INFO - Executing: 
    INSERT INTO TABLE rdv.sat_employee_job_details
    SELECT DISTINCT
        md5(so.employeenumber::text),
        'csv',
        now(),
        case when so.attrition = 'Yes' then true else false end,
        so.hourlyrate,
        so.joblevel,
        so.standardhours,
        so.stockoptionlevel
    FROM
        staging.stg_table so
    LEFT OUTER JOIN rdv.sat_employee_job_details sat ON (sat.employee_hash_key = md5(so.employeenumber::text) AND sat.created_at IS NULL)
    where 
        coalesce(case when so.attrition = 'Yes' then true else false end, false) != coalesce(sat.attrition, false) or
        coalesce(so.joblevel, 0) != coalesce(sat.job_level, 0) or
        coalesce(so.hourlyrate, 0) != coalesce(sat.hourly_rate, 0) or
        coalesce(so.standardhours, 0) != coalesce(sat.standard_hours, 0) or
        coalesce(so.stockoptionlevel, 0) != coalesce(sat.stock_option_level, 0)
[2023-10-23T17:47:06.920+0000] {base.py:73} INFO - Using connection ID 'stg_conn' for task execution.
[2023-10-23T17:47:06.923+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/models/connection.py:374: AirflowProviderDeprecationWarning: The "schema" arg has been renamed to "database" as it contained the database name.Please use "database" to set the database name.
  return hook_class(**{hook.connection_id_attribute_name: self.conn_id}, **hook_params)

[2023-10-23T17:47:06.929+0000] {base.py:73} INFO - Using connection ID 'stg_conn' for task execution.
[2023-10-23T17:47:07.018+0000] {sql.py:418} INFO - Running statement: 
    INSERT INTO TABLE rdv.sat_employee_job_details
    SELECT DISTINCT
        md5(so.employeenumber::text),
        'csv',
        now(),
        case when so.attrition = 'Yes' then true else false end,
        so.hourlyrate,
        so.joblevel,
        so.standardhours,
        so.stockoptionlevel
    FROM
        staging.stg_table so
    LEFT OUTER JOIN rdv.sat_employee_job_details sat ON (sat.employee_hash_key = md5(so.employeenumber::text) AND sat.created_at IS NULL)
    where 
        coalesce(case when so.attrition = 'Yes' then true else false end, false) != coalesce(sat.attrition, false) or
        coalesce(so.joblevel, 0) != coalesce(sat.job_level, 0) or
        coalesce(so.hourlyrate, 0) != coalesce(sat.hourly_rate, 0) or
        coalesce(so.standardhours, 0) != coalesce(sat.standard_hours, 0) or
        coalesce(so.stockoptionlevel, 0) != coalesce(sat.stock_option_level, 0), parameters: None
[2023-10-23T17:47:07.043+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/common/sql/operators/sql.py", line 280, in execute
    output = hook.run(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/common/sql/hooks/sql.py", line 392, in run
    self._run_command(cur, sql_statement, parameters)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/common/sql/hooks/sql.py", line 423, in _run_command
    cur.execute(sql_statement)
psycopg2.errors.SyntaxError: syntax error at or near "TABLE"
LINE 2:     INSERT INTO TABLE rdv.sat_employee_job_details
                        ^

[2023-10-23T17:47:07.052+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=update_dwh, task_id=sat_employee_job_details_update, execution_date=20231023T174402, start_date=20231023T174706, end_date=20231023T174707
[2023-10-23T17:47:07.074+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 232 for task sat_employee_job_details_update (syntax error at or near "TABLE"
LINE 2:     INSERT INTO TABLE rdv.sat_employee_job_details
                        ^
; 4428)
[2023-10-23T17:47:07.118+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2023-10-23T17:47:07.145+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
